modules/services/views/src/main/java/au/com/shiftyjelly/pocketcasts/views/fragments/BaseFragment.kt

```diff
@@ -94,6 +118,22 @@ open class BaseFragment : Fragment(), CoroutineScope, HasBackstack {
         if (toolbarColors != null) {
             updateStatusBarColor(toolbarColors.backgroundColor)
         }
+
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.CREATED) {
+                viewModel.uiState.collect { state ->
+                    if (profileButton is ProfileButton.Shown) {
+                        toolbar.viewTreeObserver.addOnGlobalLayoutListener(object : ViewTreeObserver.OnGlobalLayoutListener {
+                            override fun onGlobalLayout() {
+                                toolbar.updateProfileMenuImage(state.signInState)
+                                toolbar.updateProfileMenuBadge(state.showBadgeOnProfileMenu)
+                                toolbar.viewTreeObserver.removeOnGlobalLayoutListener(this)
+                            }
+                        })
```

Comment:
Could we move it to a `doOnGlobalLayout()` extension in `ViewExtensions` file? It could work similarly to `doOnLayout()` from ktx.